src/CMakeLists.txt

```diff
@@ -0,0 +1,370 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
The cmake file should be more structured, currently everything is a bit all over the place. You can define functions, e.g. `add_hdf5_dep(target)` which adds headers and linking options. 
Similarly for slow5, tensorflow etc.

Reply:
Also provide a README that shows how to build the project, and how to switch on/off various options.

Reply:
Done.